### PR TITLE
#187419014 Build out Rest points for admin shall view all acount and set  roles 

### PR DESCRIPTION
**Pull request by Tuyisenge2**
#### What does this PR do?
implement  rest  points admin shall navigate  to view all acount  and set roles

#### Description of Task to be completed?
1.build page where admin view  all  created account
2.build page  where can customize the roles users

#### How should this be manually tested?
1.first login as an admin 
2.navigate to  dashboard/admins
3.navigate  to users as dashboard/admins/users

### CODE REVIEW

**Review by JacquelineTuyisenge — APPROVED** [review]

**Review by Hakizimana-Clement — APPROVED** [review]

**Review by Hakizimana-Clement — APPROVED** [review]

**Review by JacquelineTuyisenge — APPROVED** [review]

File: src/@types/roleType.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export interface roleTypes {
+	id?: string;
+	roleName?: string;
```
Comment (leandreAlly) [inline src/@types/roleType.ts]: check well we have two file, one is "roletype.ts" and "roleTypes.ts" simplify them in one.

File: src/@types/roleTypes.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export interface roleType {
+	id?: string;
+	roleName?: string;
+}
```
Comment (leandreAlly) [inline src/@types/roleTypes.ts:4]: remove this

File: src/components/DashboardSideNav.tsx [hunk 3]
```diff
@@ -1,9 +1,8 @@
 import { Triangle } from 'lucide-react';
 import { ReactNode, useState } from 'react';
-import { appLogo } from '../utils/images';
 import Button from './buttons/Button';
+import { appLogo } from '../utils/images';
 import { dashProfileIcon } from '../utils/images';
-
 const DashboardSideNav = ({
 	children,
 	otherStyles,
```
Comment (leandreAlly) [inline src/components/DashboardSideNav.tsx:8]: inside this dashboard sideNva I need to the username to be dynamic and same case for profile picture.

File: src/redux/features/AssignrolesSlice.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+const initialState: rolesState = {
+	data: [],
+	isLoading: false,
+	error: null,
+};
+export const assignRoles = createAsyncThunk(
+	'assignRoles',
+	async (
+		{ id, role, token }: { id: string; role: string; token: string },
+		{ rejectWithValue },
```
Comment (leandreAlly) [inline src/redux/features/AssignrolesSlice.ts]: to make code more readable make create separate interface for this params you have passed into asignRoles action

File: test/pages/admin/DashboardGetUser.test.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+describe('Dashboard getuser page', () => {
+	const userData: userTYpe[] = [];
+	const roleData: roleType[] = [];
+
+	beforeAll(() => {
+		[1].map((item) => {
+			const role = db.roles.create({ roleName: 'ADMIN' });
+			roleData.push(role);
+			console.log(item);
```
Comment (leandreAlly) [inline test/pages/admin/DashboardGetUser.test.tsx]: remove this console.log()

File: test/pages/admin/EdituserRole.test.tsx [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+		await renderComponent();
+		//const editUsersHeader = screen.getByText(/edit users/i);
+		const backButton = screen.getByRole('button', { name: /back/i });
+		expect(backButton).toBeInTheDocument();
+		await userEvent.click(backButton);
+		//await waitForElementToBeRemoved(editUsersHeader);
+		//	expect(editUsersHeader).not.toBeInTheDocument();
+		//screen.debug();
+	});
+});
```
Comment (leandreAlly) [inline test/pages/admin/EdituserRole.test.tsx:191]: if you're not using this commented code, remove them or keep them.

**Review by leandreAlly — CHANGES_REQUESTED** [review]

File: src/components/DashboardSideNav.tsx (hunk 3 above)
Comment (Tuyisenge2) [reply src/components/DashboardSideNav.tsx:8]: on this we issue we proposed that we will use user profile to update the picture and username so we have to wait for  michelle's task to be merged and prepare the fix for it  @leandreAlly

**Pull request closed without merging**